types/react-tracking/index.d.ts

```diff
@@ -0,0 +1,72 @@
+// Type definitions for react-tracking 4.2
+// Project: https://github.com/NYTimes/react-tracking
+// Definitions by: Eloy Durán <https://github.com/alloy>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+
+import * as React from "react";
+
+export interface TrackingProp {
+    trackEvent({}): any;
+
+    /**
+     * This method returns all of the contextual tracking data up until this point in the component hierarchy.
+     */
+    getTrackingData(): {};
+}
+
+type Falsy = false | null | undefined | "";
+
+interface Options<T> {
+    /**
+     * By default, data tracking objects are pushed to `window.dataLayer[]`. This is a good default if you use Google
+     * Tag Manager. You can override this by passing in a dispatch function as a second parameter to the tracking
+     * decorator `{ dispatch: fn() }` on some top-level component high up in your app (typically some root-level
+     * component that wraps your entire app).
+     */
+    dispatch?(data: T): any;
+
+    /**
+     * To dispatch tracking data when a component mounts, you can pass in `{ dispatchOnMount: true }` as the second
+     * parameter to `@track()`. This is useful for dispatching tracking data on "Page" components.
+     *
+     * If you pass in a function, the function will be called with all of the tracking data from the app's context when
+     * the component mounts. The return value of this function will be dispatched in `componentDidMount()`. The object
+     * returned from this function call will be merged with the context data and then dispatched. A use case for this
+     * would be that you want to provide extra tracking data without adding it to the context.
+     */
+    dispatchOnMount?: boolean | ((contextData: T) => T);
+
+    /**
+     * When there's a need to implicitly dispatch an event with some data for every component, you can define an
+     * `options.process` function. This function should be declared once, at some top-level component. It will get
+     * called with each component's tracking data as the only argument. The returned object from this function will be
+     * merged with all the tracking context data and dispatched in `componentDidMount()`. If a falsy value is returned
+     * (`false`, `null`, `undefined`, ...), nothing will be dispatched.
+     *
+     * A common use case for this is to dispatch a `pageview` event for every component in the application that has a
+     * `page` property on its `trackingData`.
+     */
+    process?(ownTrackingData: T): T | Falsy;
```

Comment:
TIL that Falsy is a type.

Reply:
Nah, it’s defined on line 18.

Reply:
Ah, sneaky :D

Reply:
I wish it were! :D Maybe we can start our own ‘lib’, also include Flow’s `Maybe<T>` etc?

Reply:
TBH, both of them feel like they can be represented as`T?` - looks like it was added, then reverted from https://github.com/Microsoft/TypeScript/pull/7140#issuecomment-188522264 - but that's probably my Swift experience coming through